stream-coordinator/src/main/java/org/apache/kylin/stream/coordinator/coordinate/BuildJobSubmitter.java

```diff
@@ -165,6 +166,11 @@ void doRun() {
             for (StreamingCubeInfo cubeInfo : coordinator.getEnableStreamingCubes()) {
                 List<String> segmentList = checkSegmentBuildJobFromMetadata(cubeInfo.getCubeName());
                 for (String segmentName : segmentList) {
+                    boolean isFull = DefaultScheduler.getInstance().getFetcherRunner().isJobPoolFull();
+                    if (isFull) {
+                        logger.info("Hit MaxConcurrentJobLimit, hold {}.", segmentName);
+                        break;
+                    }
```

Comment:
IMO, it can add this check logical code at the beginning of function 'submitSegmentBuildJob'

Reply:
OK, it make sense.